### PR TITLE
CB-10873 Avoid crash due to usage of uninitialized variable when writ…

### PR DESCRIPTION
**Pull request by omefire**
…ing geolocation data to image destination

### CODE REVIEW

**Comment by omefire** [comment]
What happens is that when trying to write the image source, we do it with a null/nil variable ```self.data```:
https://github.com/omefire/cordova-plugin-camera/blob/CB-10873/src/ios/CDVCamera.m#L669-L671

**Comment by omefire** [comment]
For more details, see: https://issues.apache.org/jira/browse/CB-10873

File: src/ios/CDVCamera.m [hunk 1]
```diff
@@ -358,6 +358,7 @@ - (NSData*)processImage:(UIImage*)image info:(NSDictionary*)info options:(CDVPic
                 // use image unedited as requested , don't resize
                 data = UIImageJPEGRepresentation(image, 1.0);
             } else {
+                data = UIImageJPEGRepresentation(image, [options.quality floatValue] / 100.0f);
                 if (options.usesGeolocation) {
```
Comment (jasongin) [inline src/ios/CDVCamera.m]: When using the "unedited" image above, the geolocation option is ignored -- is that correct behavior? Or should the if (options.usesGeolocation) be moved to outside/after the else block?

**Comment by nikhilkh** [comment]
@alsorokin Is it possible for our appium test cover this scenario? Sounds like you will have to enable this preference to see this issue.

**Comment by omefire** [comment]
I agree on an appium test to cover this scenario, but I think it should go further than just making sure no crash happens and also validate that the EXIF header is correctly written: 
- https://github.com/gomfunkel/node-exif

I have the feeling that code path might not be well tested/exercised.

File: src/ios/CDVCamera.m (hunk 1 above)
Comment (omefire) [inline src/ios/CDVCamera.m]: @jasongin I suspect it might not be correct behavior.

**Comment by jasongin** [comment]
LGTM

**Pull request closed without merging**